agreement/proposal.go

```diff
@@ -238,14 +243,18 @@ func proposalForBlock(address basics.Address, vrf *crypto.VRFSecrets, ve Validat
 	}
 
 	ve = ve.WithSeed(newSeed)
-	proposal := makeProposal(ve, seedProof, period, address)
+	uncompressedProposal := makeProposal(ve, seedProof, period, address)
+	var compressedProposal proposal
+	compressedProposal.unauthenticatedProposal = uncompressedProposal.Compressed()
+	compressedProposal.ve = uncompressedProposal.ve
+
 	value := proposalValue{
 		OriginalPeriod:   period,
 		OriginalProposer: address,
-		BlockDigest:      proposal.Block.Digest(),
-		EncodingDigest:   crypto.HashObj(proposal),
+		BlockDigest:      compressedProposal.Block.Digest(),
+		EncodingDigest:   crypto.HashObj(compressedProposal),
 	}
-	return proposal, value, nil
+	return uncompressedProposal, value, nil
```

Comment:
this looks confusing - why do we include compressed hash but return non-compressed. Please comment out the code

Reply:
thanks for suggestion, tried to make this cleaner + added comment